### PR TITLE
[KBFS-2014] Add new traceLogger type

### PR DESCRIPTION
**Pull request by akalin-keybase**
Use it to add tracing for remote BlockServer
and MDServer. Also add tracing to BlockOps.

### CODE REVIEW

**Comment by akalin-keybase** [comment]
Running kbfsfuse with tracing again verifies that there isn't any MDServer requests left on fuse requests.

File: libkbfs/mdserver_remote.go [hunk 1]
```diff
@@ -441,11 +440,22 @@ func (md *MDServerRemote) signalObserverLocked(observerChan chan<- error, id tlf
...
 	handle *tlf.Handle, bid BranchID, mStatus MergeStatus,
-	start, stop MetadataRevision) (tlf.ID, []*RootMetadataSigned, error) {
+	start, stop MetadataRevision) (tlfID tlf.ID, rmdses []*RootMetadataSigned, err error) {
 	// figure out which args to send
 	if id == tlf.NullID && handle == nil {
 		panic("nil tlf.ID and handle passed into MDServerRemote.get")
 	}
+	var reqStr string
+	if id != tlf.NullID {
+		reqStr = fmt.Sprintf("%s", id)
```
Comment (strib) [inline libkbfs/mdserver_remote.go]: Ideally we'd only do these `Sprintf`s if tracing is on.  You could build a simple class that only computes this once `String()`is called, and then pass that to `LazyTrace`?  Or just pass both to `LazyTrace` and let the wrong one print as the zero value?  Or call `LazyTrace` inside the `if` statements? Not a big deal, but would be nice.

File: libkbfs/mdserver_remote.go [hunk 2]
```diff
@@ -441,11 +440,22 @@ func (md *MDServerRemote) signalObserverLocked(observerChan chan<- error, id tlf
...
+	} else {
+		reqStr = fmt.Sprintf("%s", handle)
+	}
+	md.log.LazyTrace(ctx, "MDServer: get %s %s %d-%d", reqStr, bid, start, stop)
+	defer func() {
+		md.log.LazyTrace(ctx, "MDServer: get %s %s %d-%d done (err=%v)", reqStr, bid, start, stop, err)
```
Comment (strib) [inline libkbfs/mdserver_remote.go]: Maybe we should have a `deferLog`?

File: libkbfs/mdserver_remote.go (hunk 1 above)
Comment (akalin-keybase) [inline libkbfs/mdserver_remote.go]: Done.

File: libkbfs/mdserver_remote.go (hunk 2 above)
Comment (akalin-keybase) [inline libkbfs/mdserver_remote.go]: Done.

**Pull request merged**